### DISCUSSION TITLE
Unable to layout item when the document contains large tables

### DISCUSSION THREAD

**Discussion by heiwilOS** [Mogztter/asciidoctor-web-pdf#405] [Q&A]
Hi guys,
when I try to convert the attached document, I get the error `Unable to layout item:  JSHandle@node`, the contained table is not shown. The attached pdf file shows the output of asciidoctor-pdf.

[documentation.zip](https://github.com/Mogztter/asciidoctor-web-pdf/files/5958909/documentation.zip)

[documentation.pdf](https://github.com/Mogztter/asciidoctor-web-pdf/files/5958907/documentation.pdf)

Heiko

**Comment by ggrossetie** [comment]
Hi Heiko,

This is a limitation of the underlying engine [Paged.js](https://www.pagedjs.org/) when working on "complex" tables.
The engine is not capable of chunking a cell with a rowspan.

The problem is that your table does not fit on one page AND the last row has a rowspan. Since Paged.js is not able to chunk the last row, the table cannot fit on the page.
The chunker aborts and leaves this element (the whole table) out of the page, hence the error "Unable to layout item".

You can workaround this issue by removing the rowspan on the last row:

```
|Umgebungsbedingungen 
|Temperatur 
|°C 
|10-45°C

|Umgebungsbedingungen 
|Feuchtigkeit 
|% 
|max. 90%

|Umgebungsbedingungen 
|Beleuchtung 
|Lux 
|250

|Umgebungsbedingungen 
|Aufstellungshöhe 
|m 
|1000
```

You can also use a small font to fit the table on one page, or manually chunk the table before converting to PDF.

**Reply by ggrossetie** [reply]
> This is a limitation of the underlying engine Paged.js when working on "complex" tables.
The engine is not capable of chunking a cell with a rowspan.

Apparently, this is not the case here, Paged.js can layout the table if you change the first line from a section title to a document title:

```adoc
= Technische Daten

// ...
```

I think Paged.js might not be able to resolve conflicting rules, a section title has a rule to avoid page break after and a table has a rule to avoid page break inside.
The chunker is not able to decide which rule should be ignored. In other words, the chunker is trying to avoid a page break after the section title and at the same time it's trying to avoid a page break inside the table. Since it's impossible, it aborts.

You can use a custom style to lift this rule:

```css
table {
  break-inside: auto;
}
```

The quick and dirty way is to use a passthrough (I recommend to only use it to troubleshoot).
Add the following line in your document:

```
++++
<style>
table {
  break-inside: auto;
}
</style>
++++
```

**Reply by ggrossetie** [reply]
I noticed that the section title "Technische Daten" is shown twice when the AsciiDoc document does not contain a title.
Please open an issue so we can fix this issue.

**Reply by heiwilOS** [reply]
looks like you're right, Paged.js might not be able to resolve conflicting rules. 
If I place any text between the section title and the table, he is showing the table.

But I can't reproduce the issue, that the section title is shown twice, when the document doesn't contain a title.

**Reply by ggrossetie** [reply]
For reference, here's the upstream merge request/issue: https://gitlab.pagedmedia.org/tools/pagedjs/merge_requests/137#note_2499

The specification says that `break-after: avoid;` has a lower precedence, so if the `<table>` elment can fit alone on a page then you will have the title on one page and the table on the other.
However, if the table does not fit alone on a page then the specification says that we should also ignore `break-inside: avoid`. So you will have the title on one page followed by the table (only a fragment) and the rest of the table on the next page (or pages depending on how long the table is).
